### PR TITLE
Improve french.lbx

### PR DESCRIPTION
**Pull request by dpled**
In accordance to #1022, I’ve modified the french.lbx file. Changes are :
− "§" have been corrected (in french it means "paragraph", not "section") ;
− "Liste des sigles" has been replaced by "Liste des abréviations" ;
− Abbreviations for "mars" and "juillet" have been corrected ;
− As mentioned in #552 and #555, French texts should always use italic for latin forms (it’s recommended by both Imprimerie nationale and Jean-Pierre Lacroux). As suggested by @moewew in #555, I use _\mkbibemph_ to do so.

### CODE REVIEW

File: tex/latex/biblatex/lbx/french.lbx [hunk 1]
```diff
@@ -1,7 +1,20 @@
 \ProvidesFile{french.lbx}
 [\abx@lbxid]
 
-% Translations by non-native speaker: this file needs an overhaul!
+\DefineBibliographyStrings{french}{%
```
Comment (moewew) [inline tex/latex/biblatex/lbx/french.lbx:4]: Is this change intended?

Comment (dpled) [reply tex/latex/biblatex/lbx/french.lbx:4]: No, in fact it’s a part of the content of my version of the traduction (for the new package), that were in my clipboard… I don’t know why, but some days ago my computer had a weird trip and has assign the middle-button of the mouse to the "paste" function… So sometimes, when I scroll, it pastes things anywhere.

Comment (moewew) [reply tex/latex/biblatex/lbx/french.lbx]: Ah. Thanks.

Comment (dpled) [reply tex/latex/biblatex/lbx/french.lbx]: I’ve corrected it in my new commit.

**Comment by moewew** [comment]
@dbitouze @maieul Any comments about the proposed change?

I'm not quite sure how I feel about the `\mkbibemph` in all the Latin terms. If we add them, `verbose-trad`'s `\mkibid` will have the opposite effect if it *also* includes `\mkbibemph`. Currently,
```latex
\documentclass[french]{article}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}

\usepackage[style=verbose-trad1, backend=biber]{biblatex}

\renewcommand*{\mkibid}[1]{\mkbibemph{#1}}

\addbibresource{biblatex-examples.bib}


\begin{document}
Lorem\autocite{knuth:ct:a}
ipsum\autocite{knuth:ct:b}
dolor\autocite{knuth:ct:a}
dolor\autocite{knuth:ct:a}
\printbibliography
\end{document}
```
gives italic Latin forms.



See also https://github.com/plk/biblatex/issues/555.

**Comment by moewew** [comment]
Oh, scrap that. The `\mkbibemph` messes with capitalisation. Unfortunately, we can't use it for `ibidem`.

```latex
\documentclass[french]{article}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}

\usepackage[style=verbose-inote, backend=biber]{biblatex}

\DefineBibliographyStrings{french}{
  ibidem = {\mkbibemph{ibid\adddot}},
}

\addbibresource{biblatex-examples.bib}


\begin{document}
Lorem\autocite{sigfridsson}
ipsum\autocite{sigfridsson}

\printbibliography
\end{document}
```
with and without the redefinition of `ibidem`.

**Comment by maieul** [comment]
Should not be possible to have a hook when switching language, chaning `\mkibid` definition ? that will be the best solution, I think.

The change seems correct, however.

**Comment by moewew** [comment]
> Should not be possible to have a hook when switching language, chaning `\mkibid` definition ? that will be the best solution, I think.


Sure, we can can make `french.lbx` set to `\mkibid` `\mkbibemph`. After all, `french.lbx` already sets `\mkbibnamefamily` to `\textsc`. We just need something for "et al." which isn't covered by `\mkibid`.

**Comment by dpled** [comment]
I don’t know if it’s a good solution, but some forms like _in_, _et al._ or _sqq._ are actually never capitalized… so if the problem of \mkbibemph is just about capitals, I think it’s not very serious to have \mkbibemph for those strings… The real problem is with _Passim_ and _Cf._

**Comment by moewew** [comment]
Came up with https://github.com/plk/biblatex/pull/1028 to avoid having to put formatting commands into the bibstrings. Let me know what you think.

File: tex/latex/biblatex/lbx/french.lbx [hunk 2]
```diff
@@ -439,25 +439,25 @@
...
+  idempf           = {{eaedem}{eaedem}},
+  idempn           = {{eadem}{eadem}},
+  idempp           = {{eidem}{eidem}},
+  ibidem           = {{ibidem}{ibid\adddot}},
+  opcit            = {{op\adddotspace cit\adddot}{op\adddotspace cit\adddot}},
+  loccit           = {{loc\adddotspace cit\adddot}{loc\adddotspace cit\adddot}},
+  confer           = {{cf\adddot}{cf\adddot}},
+  sequens          = {{sq\adddot}{sq\adddot}},
+  sequentes        = {{sqq\adddot}{sqq\adddot}},
+  passim           = {{passim}{{pass\adddot}},
```
Comment (moewew) [inline tex/latex/biblatex/lbx/french.lbx:456]: I guess this should be
```suggestion
  passim           = {{passim}{pass\adddot}},
```

**Comment by dpled** [comment]
OK, thanks ! I did a new commit removing all \mkbibemph’s to have only the translation corrections. Don’t hesitate to ask me if you want to rework the file for #1028 !

File: tex/latex/biblatex/lbx/french.lbx (hunk 2 above)
Comment (dpled) [reply tex/latex/biblatex/lbx/french.lbx]: Oops, little mistake while removing \mkbibemph… I correct it.

File: tex/latex/biblatex/lbx/french.lbx [hunk 3]
```diff
@@ -114,7 +112,7 @@
 \DeclareBibliographyStrings{%
   bibliography     = {{Bibliographie}{Bibliographie}},
   references       = {{R\'ef\'erences}{R\'ef\'erences}},
-  shorthands       = {{Liste des sigles}{Sigles}},
+  shorthands       = {{Liste des abréviations}{Abréviations}},
```
Comment (moewew) [inline tex/latex/biblatex/lbx/french.lbx:115]: In this file we are still using macro escapes for non-ASCII chars, so instead of `é` we'd write `\'e` here.
```suggestion
  shorthands       = {{Liste des abr\'eviations}{Abr\'eviations}},
```

Comment (dpled) [reply tex/latex/biblatex/lbx/french.lbx]: Sorry…

**Comment by moewew** [comment]
Thanks. All looks good from my side of things now.

I'd like to wait for a quick word by @dbitouze before we merge.

I'll ping everyone here and in #555 once #1028 is merged, so we can get it to work with the Latin phrases in `french.lbx`.

**Comment by dbitouze** [comment]
Seems okay for me.

**Comment by moewew** [comment]
Thank you all. Squashed and merged.

**Comment by moewew** [comment]
After about nine months I decided to go ahead an merge https://github.com/plk/biblatex/pull/1028, so Latin phrases in the French translation that should be italicised can now be marked up as follows

```latex
\documentclass[british,french]{article}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}

\usepackage[style=authortitle-ibid, backend=biber]{biblatex}

\DefineBibliographyStrings{french}{
  andothers = {et\addabbrvspace al\adddot},
  ibidem    = {ibid\adddot},
}

\DefineBibliographyExtras{french}{%
  \DeclareBibstringSet{latin}{andothers,ibidem}%
  \DeclareBibstringSetFormat{latin}{\mkbibemph{#1}}}

\UndefineBibliographyExtras{french}{%
  \UndeclareBibstringSet{latin}}

\addbibresource{biblatex-examples.bib}

\begin{document}
Lorem \autocite{sigfridsson}
ipsum \autocite{sigfridsson}
dolor \autocite{aksin}
\printbibliography

\selectlanguage{british}
Lorem \autocite{sigfridsson}
ipsum \autocite{sigfridsson}
dolor \autocite{aksin}
\printbibliography
\end{document}
```
